pkg/metrics/prometheus_metrics.go

```diff
@@ -98,7 +98,7 @@ func (metricsServer PrometheusMetricServer) RecordHPAScalerError(namespace strin
 		return
 	}
 	// initialize metric with 0 if not already set
-	_, errscaler := scalerErrorsTotal.GetMetricWith(getLabels(namespace, scaledObject, scaler, scalerIndex, metric))
+	_, errscaler := scalerErrors.GetMetricWith(getLabels(namespace, scaledObject, scaler, scalerIndex, metric))
```

Comment:
This was the main reason all tests were failing. It was causing the metrics server to panic since this will always return an err. 

Prometheus' [`scalerErrorsTotal`](https://github.com/kedacore/keda/blob/af5254dd2bf8f2b70eb73ddc6428b4fb01dc2fb2/pkg/metrics/prometheus_metrics.go#L14-L22) is initialized with no labels, and initialized to 0 in the [`NewServer()`](https://github.com/kedacore/keda/blob/af5254dd2bf8f2b70eb73ddc6428b4fb01dc2fb2/pkg/metrics/prometheus_metrics.go#L77-L81) method. 

However `RecordHPAScalerError()` is trying to initialize it again with labels, then it panics since #1393 if there is an error. That call was returning 

```
Unable to write to serve custom metrics: inconsistent label cardinality
```